### ISSUE TITLE
Performance table

### ISSUE THREAD

**Issue by chapmanjacobd** [hydrargyrum/decoyfs#1]
It would be interesting to see how much faster this is compared to `ext4` or `btrfs` for browsing folders with deep hierarchies or with many files

**Comment by hydrargyrum** [comment]
Good question, I will try to do some benchmark. What tool would you recommend? bonnie++? kdiskmark? Something else?

Note however that better performance than "real" filesystems is not a goal per se, given that:
- file contents cannot be read with decoyfs (no file data is stored in decoyfs database), only file metadata
- it's originally meant as as snapshot for offline media like usb keys or CDs (like [catcli](https://github.com/deadc0de6/catcli))

**Comment by chapmanjacobd** [comment]
> better performance than "real" filesystems is not a goal per se

yes I understand that but I think it would still be very interesting to see

I'm not sure what performance tool would be best. maybe after running

sync; echo 3 > /proc/sys/vm/drop_caches 

and then run os.scandir() multiple times and compare the speed using stats like sharkdp hyperfine does

**Comment by hydrargyrum** [comment]
On a FS with ~100k entries. I used `find -ls` instead of `scandir.py`.

Without `--prepare`:
- 2-3s regular fs
- 23s decoyfs

With `--prepare`:
- 53s regular fs
- 23s decoyfs

Fuse has a few caching options, but adding them didn't change anything.

Then I innocently added `-s` option to decoyfs to be single-threaded. decoyfs performance flew to only ~5 seconds. I have yet to understand why this unexpected change. All files were still listed exactly the same though. There's probably a bug in decoyfs' multithreading handling.

**Comment by chapmanjacobd** [comment]
5 seconds in single threaded mode with `--prepare` ? or without `--prepare` ?

**Comment by hydrargyrum** [comment]
With `--prepare`. The sqlite database file is 21 MB.
